psycopg/psycopg/errors.py

```diff
@@ -18,19 +18,210 @@
+        raise OperationalError("the connection is closed")
+
+    def send_query_prepared(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def prepare(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def exec_prepared(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def describe_prepared(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def send_describe_prepared(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def describe_portal(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def send_describe_portal(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def get_result(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def consume_input(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def is_busy(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def flush(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def set_single_row_mode(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def get_cancel(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def notifies(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def put_copy_data(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def put_copy_end(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def get_copy_data(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def trace(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def set_trace_flags(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def untrace(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def encrypt_password(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def make_empty_result(self, *args: Any) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def enter_pipeline_mode(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def exit_pipeline_mode(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def pipeline_sync(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+    def send_flush_request(self) -> NoReturn:
+        raise OperationalError("the connection is closed")
+
+
+def _finish_pgconn(pgconn: PGconn) -> PGconn:
```

Comment:
It felt cleaner to me as a function because it not only creates the `FinishedPGconn` but also does `pgconn.finish()` which would be, IMO, quite surprising for a class method.

Made it "public" though.